src/index.ml

```diff
@@ -782,6 +783,18 @@ struct
                 t.log;
               may (fun (i : index) -> IO.close i.io) t.index;
               may (fun lock -> IO.unlock lock) t.writer_lock ))
+
+  let ro_sync t =
+    Stats.incr_nb_ro_sync ();
+    let t = check_open t in
+    Log.info (fun l -> l "[%s] ro_sync" (Filename.basename t.root));
+    if t.config.readonly then sync_log t
+    else Fmt.failwith "only an RO instance should call this function"
```

Comment:
`Fmt.invalid_arg` is probably better here to indicate a user error (e.g. a wrong use of the API, not a bug in our code).